### ISSUE TITLE
seiscomp3 inventory xml

### ISSUE THREAD

**Issue by blazvicic** [obspy/obspy#1536]
Hi!

As I was told on the mailing list, I am posting this bug (?) here.

This is the code, that was supposedly read inventory xml, that I downloaded from http://eida.gfz-potsdam.de/webdc3/

The file is this one:
https://drive.google.com/open?id=0BzMcKedS7yadMkRzY0RoS2FDVFU

Code I used was this one:

```
from obspy import read_inventory

inv = read_inventory("Package_1475048499259-GFZ_73152721_inventory.xml", format = "sc3ml")
print inv
```

And the error I get: 

> Traceback (most recent call last):
>   File "<stdin>", line 1, in <module>
>   File "<decorator-gen-83>", line 2, in read_inventory
>   File "/home/blaz/miniconda2/lib/python2.7/site-packages/obspy/core/util/decorator.py", line 294, in _map_example_filename
>     return func(*args, **kwargs)
>   File "/home/blaz/miniconda2/lib/python2.7/site-packages/obspy/core/inventory/inventory.py", line 72, in read_inventory
>     format=format)[0]
>   File "/home/blaz/miniconda2/lib/python2.7/site-packages/obspy/core/util/base.py", line 464, in _read_from_plugin
>     list_obj = read_format(filename, **kwargs)
>   File "/home/blaz/miniconda2/lib/python2.7/site-packages/obspy/io/seiscomp/sc3ml.py", line 158, in _read_sc3ml
>     for net_element in inv_element.findall(_ns("network")):
> AttributeError: 'NoneType' object has no attribute 'findall'

Thanks, 
Blaž

**Comment by Jollyfant** [comment]
Hi, I will try to have a look today.

**Comment by Jollyfant** [comment]
It seems that the file format that was supplied by WebDC3 is different from the one implemented under `format="sc3ml"` in ObsPy. It could be a different (older) schema version but I'm not entirely sure. I'll add an exception when the input file is not in the expected format.

The following query gives you the inventory file and response information for the entire SL network in the supported sc3ml format:

`http://www.orfeus-eu.org/fdsnws/station/1/query?net=SL&format=sc3ml&level=response`

The following call works (preferably cache it on your drive):

    from obspy import_inventory
    inv = read_inventory("http://www.orfeus-eu.org/fdsnws/station/1/query?net=SL&format=sc3ml&level=response")

I will have to forward the question because I do not know what format is returned by WebDC3. Pinging @andres-h for some more information on this.

**Comment by megies** [comment]
@jsaul might also be able to give insight on the format details..

**Comment by jsaul** [comment]
Hi all, thanks for bringing this up. @andres-h will provide a detailed reply later today.

**Comment by megies** [comment]
> @andres-h will provide a detailed reply later today.

I guess you're referring to the [obspy-users] mailing list post? http://lists.swapbytes.de/archives/obspy-users/2016-September/002155.html

**Comment by andres-h** [comment]
Yes, I posted a longer explanation to the mailing list. To sum it up: 
InventoryXML format (ArcLink, WebDC) seems to be indeed currently not 
supported by read_inventory(), but it is (obviously) supported by 
obspy.clients.arclink.client.Client.get_inventory(), so I guess the code 
just needs some refactoring.

SC3ML, as demonstrated above, is a non-standard FDSNWS extension. It is 
generally recommended to stick to FDSNWS standard and use the FDSN 
StationXML format (until there isn't a better standard).

**Comment by Jollyfant** [comment]
> SC3ML, as demonstrated above, is a non-standard FDSNWS extension. It is generally recommended to stick to FDSNWS standard and use the FDSN StationXML format (until there isn't a better standard).

The reason `SC3ML` was implemented is because it's offered by most of the FDSN Station services besides `stationXML` and `fdsnXML`. The reason this alternative markup is powerful is because it is very information dense and may be up to two orders of magnitude smaller for an entire inventory.

It will be a good idea to take out the `inventoryXML` parsing from `obspy.clients.arclink.client.Client.get_inventory()` and stick it in `obspy.read_inventory`. I will have a look at doing that the coming week.

**Comment by krischer** [comment]
> It will be a good idea to take out the inventoryXML parsing from obspy.clients.arclink.client.Client.get_inventory() and stick it in obspy.read_inventory. I will have a look at doing that the coming week.

I agree that this would be a very good idea but its likely a lot of work as it does not parse to an `Inventory `object but a custom structure (or maybe its very similar to the sc3ml which you already implemented?).

This is historical as the arclink module precedes the `Inventory` object by several years and we never really needed it as we are just using the FDSN web services. (Same reason `obspy.io.xseed` does not parse to an `Inventory` object which, in a perfect world, it should).

**Comment by megies** [comment]
I think this can be closed, as this has to be regarded a new feature and is being addressed in #1539 for version 1.1.0. Please reopen if I'm mistaken.
